### PR TITLE
Fixes #12772: Fix -static builds

### LINKED ISSUE #12772: Openssl 1.1.1 static builds fail to produce a viable image
Starting with 1.1.1c (i.e. 1.1.1b worked fine), when trying to build a non-dynamically-linked image via the `-static` option, the executable produced (at least under linux) core dumps immediately.

```
git checkout OpenSSL_1_1_1c
./config -static
make
apps/openssl version
```

This worked fine under OpenSSL 1.1.1b. The workaround is to add some options that need to be disabled:
```
git checkout OpenSSL_1_1_1c
./config -static no-shared no-pic no-threads
make
apps/openssl version
```
It appears that the fix for #8787 (commit 007213795a0e15901cbdc4558ae2a8c21d3ad9bb) broke this, based on my git bisect. As far as I can tell, the setting of some of the `$disables{}` occurs too late with that change. I have yet to figure out the solution, but I wanted to report it before I forgot about it.

### PR DESCRIPTION
**Pull request by tmshort**
Pull in check from #10878
Move disabling of pic, threads and statics up higher before they
are checked.

### CODE REVIEW

File: Configure
Comment (levitte) [inline Configure:1204]: Are you sure the regexp is needed?  After all, you're grepping a list of words...

Comment (tmshort) [reply Configure:1204]: Admittedly, I was pulling in the regex from #10878. I will check without it.

Comment (tmshort) [reply Configure]: Looks to be not needed.

**Comment by tmshort** [comment]
Still waiting on review?

**Review by t8m — APPROVED** [review]

**Comment by tmshort** [comment]
ping @levitte ?

**Comment by tmshort** [comment]
ping @levitte ?

**Review by levitte — APPROVED** [review]
I meant to formally approve yesterday, before settings the "approved: done" label, so will count this as an approval that got delayed "for technical reasons".  This doesn't reset the grace time timer.

**Comment by levitte** [comment]
I'm merging this now

**Comment by levitte** [comment]
Merged

cfd7225fbb9507b2e443a494459bdaab5236d29d Fix -static builds

**Pull request closed without merging**
